lib/deimos/config/configuration.rb

```diff
@@ -0,0 +1,330 @@
+# frozen_string_literal: true
+
+require_relative 'phobos_config'
+require_relative 'configurable'
+require_relative '../metrics/mock'
+require_relative '../tracing/mock'
+
+# :nodoc:
+module Deimos
+  include Configurable
+
+  # :nodoc:
+  class Configurable::ConfigStruct
+    include Deimos::PhobosConfig
+  end
+
+  # :nodoc:
+  after_configure do
+    Phobos.configure(self.config.phobos_config)
+    self.config.producer_objects.each do |producer|
+      configure_producer_or_consumer(producer)
+    end
+    self.config.consumer_objects.each do |consumer|
```

Comment:
It shouldn't be, since we've moved everything to a single config file. I'm not sure what the goal would be of having them. If the issue is that there's a consumer with a topic that isn't encoded, one of the things on the roadmap would be to specify different backends for topics (e.g. using Avro without a schema registry, JSON schema, no encoding at all, Protobuf etc.) which would allow this behavior.